clang-tools-extra/clang-tidy/misc/ConstCorrectnessCheck.cpp

```diff
@@ -39,34 +41,47 @@ ConstCorrectnessCheck::ConstCorrectnessCheck(StringRef Name,
     : ClangTidyCheck(Name, Context),
       AnalyzeValues(Options.get("AnalyzeValues", true)),
       AnalyzeReferences(Options.get("AnalyzeReferences", true)),
+      AnalyzePointers(Options.get("AnalyzePointers", true)),
       WarnPointersAsValues(Options.get("WarnPointersAsValues", false)),
+      WarnPointersAsPointers(Options.get("WarnPointersAsPointers", true)),
       TransformValues(Options.get("TransformValues", true)),
       TransformReferences(Options.get("TransformReferences", true)),
       TransformPointersAsValues(
           Options.get("TransformPointersAsValues", false)),
+      TransformPointersAsPointers(
+          Options.get("TransformPointersAsPointers", true)),
       AllowedTypes(
           utils::options::parseStringList(Options.get("AllowedTypes", ""))) {
-  if (AnalyzeValues == false && AnalyzeReferences == false)
+  if (AnalyzeValues == false && AnalyzeReferences == false &&
+      AnalyzePointers == false)
     this->configurationDiag(
         "The check 'misc-const-correctness' will not "
-        "perform any analysis because both 'AnalyzeValues' and "
-        "'AnalyzeReferences' are false.");
+        "perform any analysis because both 'AnalyzeValues', "
```

Comment:
```suggestion
        "perform any analysis because 'AnalyzeValues', "
```
Since there are 3 items now